### ISSUE TITLE
wrapper updates for libpd 0.12.0

### ISSUE THREAD

**Issue by danomatika** [libpd/libpd#317]
Pinging @nettoyeurny @Ant1r @chr15m @residuum @joebowbeer

There are a few updates to the libpd C API for the 0.12.0 release. I noticed today that not all the wrappers are in sync yet, so I'm adding the newer functions to the C++ API. Can some of you take a look and update the other language wrappers?

* [x] Obj-C: up to date, more or less
* [x] C++: updating this now... **done**
* [ ] C#: this was more recently updated, so should only need the new process & array resize functions
* [ ] Java: this hasn't been updated in a while...
* [x] Python: this should be ok as I updated the SWIG interface recently for Python 3, someone could double-check

Off the top of my head, recent additions to the C API are:

* libpd_process_raw_short
* libpd_process_raw_double
* libpd_array_resize
* any other API additions for libpd 0.11 and prev: gui, instance, verbose, etc

Depending on the language, not *everything* needs to be wrapped, ie. instance functions.

Once these are in place, we can put out a 0.12.1 release.

**Comment by fwestberg** [comment]
Hi Dan,

I am trying to use libpd and the Python wrapper on a Raspberry Pi 4. I am running Debian GNU/Linux 10 (buster) and Python 3.7.3 in a Docker container.

I have checked out and built version 0.12.1.
The installation process for the Python wrapper completes with a couple of warnings, but I can run the scripts in /samples/python successfully.

However, as soon as I try to implement audio, things go wrong.

```bash
TypeError: argument 1 must be read-only bytes-like object, not array.array
Traceback (most recent call last):
  File "test.py", line 27, in <module>
    time.sleep(.1)
TypeError`
```

The traceback rarely makes sense, but I think I've narrowed the problem down to the input argument for `PdManager.process()`

I am running the script and patch attached below (which I have been using successfully with Python 2 before):
[libpd-test-code.zip](https://github.com/libpd/libpd/files/5998369/libpd-test-code.zip)

Let me know, if I should supply further information (or create a separate issue).

All the best,
Flemming
